samza-core/src/main/java/org/apache/samza/zk/ZkKeyBuilder.java

```diff
@@ -88,5 +88,7 @@ public String getJobModelPath(String jobModelVersion) {
     return String.format("%s/%s", getJobModelPathPrefix(), jobModelVersion);
   }
 
-
+  public String getJobModelVersionBarrierPrefix() {
+    return String.format("/%s/versionBarriers", pathPrefix);
```

Comment:
when you use pathPrefix, does it already have a "/" at the beginning? If so, you may not want the additional "/"